### PR TITLE
New single balance ReadPool

### PR DESCRIPTION
**Pull request by iamrz1**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:https://github.com/0chain/blobber/pull/723
- gosdk: https://github.com/0chain/gosdk/pull/486
- zbox-cli: https://github.com/0chain/zboxcli/pull/256
- system_test: https://github.com/0chain/system_test/pull/297

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | single-balance-readpool  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | single-balance-readpool  |
| zwalletcli  | staging  |
| 0chain  | single-balance-readpool  |
| blobber  | single-balance-readpool  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | single-balance-readpool  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | single-balance-readpool  |
| zwalletcli  | staging  |
| 0chain  | single-balance-readpool  |
| blobber  | single-balance-readpool  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/readpool.go [hunk 1]
```diff
@@ -76,22 +74,13 @@ func (rp *readPool) Decode(p []byte) error {
...
 }
 
-func (rp *readPool) add(isOwner bool, balance currency.Coin) {
-	if isOwner {
-		rp.OwnerBalance += balance
-		return
-	}
-	rp.VisitorBalance += balance
+func (rp *readPool) add(balance currency.Coin) {
+	rp.Balance += balance
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/readpool.go]: Call currency.AddCoin() to add coins and check error

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/readpool.go (hunk 1 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/storagesc/readpool.go]: done

File: code/go/0chain.net/smartcontract/storagesc/benchmark_tests.go [hunk 2]
```diff
@@ -522,13 +522,7 @@ func BenchmarkTests(
 				ToClientID:   ADDRESS,
 				CreationDate: creationTime,
 			},
-			input: func() []byte {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/benchmark_tests.go:525]: probably we need empty struct here, have you ran benchmarks with these changes?

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/benchmark_tests.go (hunk 2 above)
Comment (iamrz1) [reply code/go/0chain.net/smartcontract/storagesc/benchmark_tests.go:525]: check done.

**Review by dabasov — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**